compiler-core/src/pretty.rs

```diff
@@ -41,73 +42,79 @@ pub trait Documentable<'a> {
 
 impl<'a> Documentable<'a> for char {
     fn to_doc(self) -> Document<'a> {
-        Document::String(format!("{self}"))
+        Document::string(format!("{self}"))
+    }
+}
+
+impl<'a> Documentable<'a> for String {
+    fn to_doc(self) -> Document<'a> {
+        Document::string(self)
     }
 }
 
 impl<'a> Documentable<'a> for &'a str {
     fn to_doc(self) -> Document<'a> {
-        Document::Str(self)
+        Document::str(self)
     }
 }
 
 impl<'a> Documentable<'a> for EcoString {
     fn to_doc(self) -> Document<'a> {
-        Document::EcoString(self)
+        Document::eco_string(self)
     }
 }
 
 impl<'a> Documentable<'a> for &EcoString {
     fn to_doc(self) -> Document<'a> {
-        Document::EcoString(self.clone())
+        Document::eco_string(self.clone())
     }
 }
 
 impl<'a> Documentable<'a> for isize {
     fn to_doc(self) -> Document<'a> {
-        Document::String(format!("{self}"))
+        Document::string(format!("{self}"))
     }
 }
 
 impl<'a> Documentable<'a> for i64 {
     fn to_doc(self) -> Document<'a> {
-        Document::String(format!("{self}"))
+        Document::string(format!("{self}"))
     }
 }
 
 impl<'a> Documentable<'a> for usize {
     fn to_doc(self) -> Document<'a> {
-        Document::String(format!("{self}"))
+        Document::string(format!("{self}"))
     }
 }
 
 impl<'a> Documentable<'a> for f64 {
     fn to_doc(self) -> Document<'a> {
-        Document::String(format!("{self:?}"))
+        Document::string(format!("{self:?}"))
     }
 }
 
 impl<'a> Documentable<'a> for u64 {
     fn to_doc(self) -> Document<'a> {
-        Document::String(format!("{self:?}"))
+        Document::string(format!("{self:?}"))
     }
 }
 
 impl<'a> Documentable<'a> for u32 {
     fn to_doc(self) -> Document<'a> {
-        Document::String(format!("{self}"))
+        Document::string(format!("{self}"))
     }
 }
 
 impl<'a> Documentable<'a> for u16 {
     fn to_doc(self) -> Document<'a> {
-        Document::String(format!("{self}"))
+        Document::string(format!("{self}"))
     }
 }
 
 impl<'a> Documentable<'a> for u8 {
     fn to_doc(self) -> Document<'a> {
-        Document::String(format!("{self}"))
+        Document::string(format!("{self}"))
```

Comment:
Could you make all these formats into eco_format please. 🙏  Ideally we remove String support altogether